### PR TITLE
Add docs for saturating integer arithmetic.

### PR DESCRIPTION
**Pull request by jbcrail**
This was added for #23241.

### CODE REVIEW

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -345,6 +345,16 @@ pub trait Int
 
     /// Saturating integer addition. Computes `self + other`, saturating at
     /// the numeric bounds instead of overflowing.
+    ///
+    /// # Example
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: These should be 'examples'

File: src/libcore/num/mod.rs [hunk 2]
```diff
@@ -345,6 +345,16 @@ pub trait Int
...
+    ///
+    /// ```rust
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: you don't need the `rust` for API documentation

**Comment by steveklabnik** [comment]
r=me with two nits, thanks!

**Comment by jbcrail** [comment]
Thanks for the comments. I've updated the branch.

On a related note, I noticed other examples have the same issues that you mentioned. Would it be too pedantic if I corrected those in another pull request?

**Comment by steveklabnik** [comment]
I've been trying to play 'boy scout rule' with them, but if you'd like to clean them up, I'd love that.

**Comment by steveklabnik** [comment]
oh, and would you mind squashing these please? thanks

**Comment by jbcrail** [comment]
I squashed the commits. I also sent a pull request to remove the Rust syntax highlighting from the example docs.

**Comment by jbcrail** [comment]
Is anything else needed to close this pull request? Thanks.

**Comment by steveklabnik** [comment]
Whoops, sorry about that! Sometimes these things slip. Thank you!

**Pull request merged**